src/backend/access/appendonly/appendonlywriter.c

```diff
@@ -778,6 +778,53 @@ RegisterSegnoForCompactionDrop(Oid relid, List *compactedSegmentFileList)
 	return;
 }
 
+void
+UpdateSegnoAfterCompaction(Oid relid, List *compactedSegmentFileList)
+{
+	AORelHashEntryData *aoentry;
+	Relation	aosegrel;
+	bool	   *awaiting_drop = NULL;
+	int			i;
+
+	Assert(Gp_role != GP_ROLE_EXECUTE);
+	if (Gp_role == GP_ROLE_UTILITY)
+	{
+		return;
+	}
+
+	if (compactedSegmentFileList == NIL)
+	{
+		return;
+	}
+
+	aosegrel = relation_open(relid, AccessShareLock);
+	awaiting_drop = get_awaiting_drop_status_from_segments(aosegrel);
+	relation_close(aosegrel, AccessShareLock);
+
+	acquire_lightweight_lock();
+
+	aoentry = AORelGetOrCreateHashEntry(relid);
+	Assert(aoentry);
+
+	for (i = 0; i < MAX_AOREL_CONCURRENCY; i++)
+	{
+		AOSegfileStatus *segfilestat = &aoentry->relsegfiles[i];
```

Comment:
The segfilestat variable is used only one time or never if condition below is false. I suggest to remove this variable and write "aoentry->relsegfiles[i].state = DROP_USE"

Reply:
I just copied RegisterSegnoForCompactionDrop and modified a bit.

Reply:
In RegisterSegnoForCompactionDrop the segfilestat variable is used two times. We have no goal to write a function which will be as similar as possible to RegisterSegnoForCompactionDrop

Reply:
fixed